scanamo/src/main/scala/com/gu/scanamo/query/ConditionExpression.scala

```diff
@@ -58,6 +58,28 @@ object ConditionExpression {
     }
   }
 
+  implicit def symbolValueInCondition[V: DynamoFormat] = new ConditionExpression[(Symbol, Set[V])] {
+    override def apply(pair: (Symbol, Set[V]))(condition: Option[RequestCondition]): RequestCondition = {
+      attributeValueInCondition.apply((AttributeName.of(pair._1), pair._2))(condition)
+    }
+  }
+
+  implicit def attributeValueInCondition[V: DynamoFormat] = new ConditionExpression[(AttributeName, Set[V])] {
+    val prefix = "inCondition"
+    override def apply(pair: (AttributeName, Set[V]))(condition: Option[RequestCondition]): RequestCondition = {
+      val format = DynamoFormat[V]
+      val attributeName = pair._1
+      val attributeValues = pair._2.zipWithIndex.map { case (v, i) => 
+        s":conditionAttributeValue$i" -> format.write(v)
+      }.toMap
+      RequestCondition(
+        s"""#${attributeName.placeholder(prefix)} IN ${attributeValues.keys.mkString("(", ",", ")")}""",
```

Comment:
Should the values be always quoted? Even if they aren't strings?

Reply:
Given a pair `'hello -> Set(1,2,3)`, this will produce `#inConditionhello IN (:conditionAttributeValue0,:conditionAttributeValue1,:conditionAttributeValue2)`. In the DynamoDB API, `#str` is a name binding expression, and `:str` is a value binding expression.

- for keys: `query.withExpressionAttributeNames(Map("#str" -> "UserId"))`
- for values, you must bind each expression to a corresponding `AttributeValue`, e.g.: `query.withExpressionAttributeValues(Map(":str" -> new AttributeValue().withN("1234")))`